ToxicR workinprogress.R

```diff
@@ -0,0 +1,51 @@
+library("devtools")
+#requires dependency Cmake
+#devtools::install_github("NIEHS/ToxicR")
+library("ToxicR")
+library("dplyr")
+library("PMCMRplus")
+library(parallel)
+library("pbmcapply")
+#bmdoutput <- read.table("~/storage/biomarker_input_.txt",
+#                        header=FALSE, sep = "\t", row.names = 1)
+bmdoutput <- read.table("~/storage/ToxicR mod/data for BMDExpress log2 CPM Males.txt",
+                        header=FALSE, sep = "\t", row.names = 1)
+#
+#bmdoutput <- bmdoutput[-3,-c(1:3,5:9,11:15,17:18)]
+#
+bmdoutput <- as.data.frame(t(bmdoutput))
+bmdoutput <-bmdoutput[,-1]
+bmdoutput[-1] <- lapply(bmdoutput[-1], function(x) {
+  as.numeric(as.character(x))
+})
+
+dose_column <-as.numeric(bmdoutput[,1]) 
+start.time <- Sys.time()
+Func <- function(resp_bygene_column){ 
+  trendtestres<- williamsTest(resp_bygene_column ~ dose_column, alternative = c("greater", "less"))
```

Comment:
@raiduhs pointed out a downside to this, we can't set alpha with this implementation

Reply:
He also gave this a test at his end and found that it might need to be run for "greater" and "less" as separate tests, can you confirm whether you get both alternative hypotheses output?

```
> williamsTest(-1*x ~ g, alternative = c("greater", "less"))
 
         Williams trend test
 
data:  -1 * x by g 
alternative hypothesis:  greater

H0
               t'-value df t'-crit decision alpha
mu1 - ctr <= 0   -1.435 12   1.782   accept  0.05
mu2 - ctr <= 0   -1.435 12   1.873   accept  0.05
---
```

```
> williamsTest(-1*x ~ g, alternative = "less")
 
         Williams trend test
 
data:  -1 * x by g 
alternative hypothesis:  less
 
H0

               t'-value df t'-crit decision alpha
mu1 - ctr >= 0    1.357 12   1.782   accept  0.05
mu2 - ctr >= 0    2.950 12   1.873   reject  0.05
---
```

Reply:
I have updated the code to test both greater and less, I am also check the ToxicR williamtrendtest and unfortunately it also does not allow for choosing an alpha value, I think we might be revamping the williamtrendtest in general anyways based on the BMDexpress source files I have sent you